File: Beginner/ATM (HS08TEST)/atm.py
w, b = input().split()              
w = int(w)                        
b = float(b)                            
if (w % 5 == 0 and b>(w+.5)):
    b = b - w - 0.5     
    print('%.2f' % b)
else:
    print('%.2f' % b)

File: Beginner/ATM (HS08TEST)/atm.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(input.readLine());

        int x = Integer.parseInt(st.nextToken());
        double y = Double.parseDouble(st.nextToken());

        if (x + 0.5 > y || x % 5 != 0) System.out.printf("%.2f", y);
        else System.out.printf("%.2f", y - x - 0.5);
    }
}